lib/active_merchant/billing/gateways/braintree/braintree_common.rb

```diff
@@ -26,6 +26,9 @@ def scrub(transcript)
 
   def supports_network_tokenization?
     response = authorize(1, ActiveMerchant::Billing::NetworkTokenizationCreditCard.test_credit_card)
+    response_code = response.params['braintree_transaction'].try!(:[], 'processor_response_code')
+    raise(ActiveMerchant::InvalidResponseError, response.message) unless response_code
```

Comment:
> What happens in case of a connection error or 5xy response and this body isn't there?

The response object is created wether or not it is successful (https://github.com/activemerchant/active_merchant/blob/98bdad6d1fe3140df43228814abcbf2429791937/lib/active_merchant/billing/gateways/braintree_blue.rb#L420). It will not raise which will cause errors where we are expecting specific params.

> It is appropriate that we let that bubble up (probably, yes)?

Yes, as far as I can tell

> I'm assuming a request error will be raised before we get to the hash traversal, yeah?
> So, was this actually necessary?

Yes, as far as I can tell, there's nothing raising when the auth fails. It just returns the response with the error code. 

> This is also the incorrect error to raise here

The params we're getting back don't match our expectation. I agree it's probably not the same expectation as other exception using this error. How about a new error type `UnexpectedResponseError`